dll/win32/shell32/shellrecyclebin/recyclebin.c

```diff
@@ -12,7 +12,7 @@ BOOL WINAPI
 CloseRecycleBinHandle(
     IN HDELFILE hDeletedFile)
 {
-    IRecycleBinFile *rbf = (IRecycleBinFile *)hDeletedFile;
+    IRecycleBinFile *rbf = IRecycleBinFileFromHDELFILE(hDeletedFile);
```

Comment:
Because that cast is nuts. At least with this change, we can try to track all of these down and settle on either HDELFILE or IRecycleBinFile in the future. The recycle bin folder is the only caller, we don't need this cumbersome abstraction.